napari/layers/labels/labels.py

```diff
@@ -262,6 +266,29 @@ def n_dimensional(self, n_dimensional):
         self._n_dimensional = n_dimensional
         self.events.n_dimensional()
 
+    @property
+    def countour(self):
+        """bool: displays contours of labels instead of shaded regions."""
+        return self._countour
+
+    @countour.setter
+    def countour(self, countour, refresh=True):
+        self._countour = countour
+        self.events.countour()
+
+        if refresh is True:
+            self._save_history()
+
+        if countour:
+            self.data = np.logical_xor(
+                self.data, ndi.binary_erosion(self.data)
+            )
+        else:
+            self.data = ndi.binary_fill_holes(self.data)
```

Comment:
My intuition is that we don't want to directly operate on the layer data, as I don't think this is perfectly reversible for all cases (i.e., if you set and then unset contours, the data may not be exactly the same). Perhaps we can apply this when the in view slice is being passed to vispy. I _think_ it might be the[ `_data_view` property on the parent Image layer](https://github.com/napari/napari/blob/c1406f67fd5570a70f5d69ae8d956ca5ceee3021/napari/layers/image/image.py#L305-L307), but I am not totally caught up on some of the recent changes to the `Image` layer. I am curious what @sofroniewn and @jni think.

Reply:
Yes, we don't want to do this on the "raw" `data`, but on the `_data_view` as @kevinyamauchi said. This will be great functionality though! Thanks for the contribution

Reply:
The way it is currently coded does indeed change the information inside `data`, I'll have a look at the `_raw_to_displayed` method.